dynamicio/mixins.py

```diff
@@ -672,28 +676,64 @@ def _read_from_postgres(self) -> pd.DataFrame:
 
         connection_string = f"postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}"
 
-        model = self.options.get("model")
         sql_query = self.options.get("sql_query")
 
         query = None
 
-        if model and sql_query:
-            raise ValueError("Only one of `model` and `sql_query` must be provided")
-        if model is None and sql_query is None:
-            raise ValueError("One of `model` and `sql_query` must be provided")
+        if self.options.get("model"):
+            raise ValueError("`model` must not be provided")
 
-        if model:
-            query = Query(self._get_table_columns(model))
+        if "schema" not in self.sources_config:
+            schema_dict = self.schema
         else:
+            schema_dict = self.sources_config["schema"]
+        schema_name = self.sources_config["name"]
+
+        model = self._generate_model_from_schema(schema_dict, schema_name)
+
+        query = Query(self._get_table_columns(model))
+        if sql_query:
             query = sql_query
 
         with session_for(connection_string) as session:
             return self._read_database(session, query, **self.options)
 
+    @staticmethod
+    def _generate_model_from_schema(schema_dict, schema_name):
+        """Generates a model from a schema loaded from yaml file.
+
+        Args:
+            schema_dict<dict>: A dictionary from source_config variable containing
+                                columns and their types
+            schema_name<string>: Name of the schema from schema yaml file.
+
+        Returns:
+            <class 'sqlalchemy.orm.decl_api.DeclarativeMeta'>
+        """
```

Comment:
Please include type annotations in function definitions and remove them from docstrings so that we are aligned with how we do it elsewhere in the project. 

```python
def _generate_model_from_schema(schema_dict: dict, schema_name: str) -> DeclarativeMeta:
     ...
```

Also please remove the docstring as this is not part of the public interface.